helpers/QtiPackageExporter.php

```diff
@@ -20,40 +20,99 @@
 
 namespace oat\taoQtiTest\helpers;
 
-use League\Flysystem\Util;
-use oat\oatbox\service\ConfigurableService;
+use common_Exception;
+use common_exception_Error;
+use common_report_Report;
+use oat\oatbox\filesystem\File;
+use oat\oatbox\filesystem\FileSystemService;
+use oat\tao\helpers\FileHelperService;
+use oat\tao\model\service\InjectionAwareService;
 use taoQtiTest_models_classes_export_TestExport22 as TestExporter;
-use core_kernel_classes_Resource as RdfResource;
-use tao_helpers_File as FileHelper;
 
-class QtiPackageExporter extends ConfigurableService
+class QtiPackageExporter extends InjectionAwareService
 {
+    public const SERVICE_ID = 'taoQtiTest/QtiPackageExporter';
+    public const QTI_PACKAGE_FILENAME = 'qti_test_export';
+
+    /** @var TestExporter */
+    private $testExporter;
+
+    /** @var FileSystemService */
+    private $fileSystemService;
+
+    /** @var FileHelperService */
+    private $fileHelperService;
+
+    public function __construct(
+        TestExporter $testExporter,
+        FileSystemService $fileSystemService,
+        FileHelperService $fileHelperService
+    ) {
+        $this->testExporter = $testExporter;
+        $this->fileSystemService = $fileSystemService;
+        $this->fileHelperService = $fileHelperService;
+    }
+
     /**
-     * @param RdfResource $test
+     * @param string $testUri
      *
      * @return array
-     * @throws \common_Exception
-     * @throws \common_exception_Error
+     * @throws common_Exception
+     * @throws common_exception_Error
      */
-    public function exportDeliveryQtiPackage(RdfResource $test)
+    public function exportDeliveryQtiPackage(string $testUri)
```

Comment:
Missing return type, and also returning with generic array is not nice

Reply:
Fixed